### PR TITLE
Stable image from StabilityAI api

### PR DESCRIPTION
**Pull request by WBlair1**
Adds Ultra, Core, and SD3 image generation.

## Checklist:

- [x] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

### CODE REVIEW

File: public/scripts/extensions/stable-diffusion/settings.html [hunk 1]
```diff
@@ -189,7 +190,82 @@
...
+                            <option value="photographic">Photographic</option>
+                            <option value="pixel-art">Pixel Art</option>
+                            <option value="tile-texture">Tile Texture</option>
+                        </select>
+                    </div>
+                </div>
+            
+                <div class="flex-container">
+                    <div class="flex1">
+                        <label for="sd_stability_aspect_ratio">Aspect Ratio</label>
```
Comment (Cohee1207) [inline public/scripts/extensions/stable-diffusion/settings.html]: Is this needed? We got a selector for width/height below everything else.

File: public/scripts/extensions/stable-diffusion/settings.html [hunk 2]
```diff
@@ -189,7 +190,82 @@
...
+                    </div>
+                    <div class="flex1">
+                        <label for="sd_stability_seed">Seed</label>
+                        <input id="sd_stability_seed" type="number" class="text_pole" value="0" min="0" max="4294967295" />
+                    </div>
+                </div>
+            
+                <div class="flex-container">
+                    <div class="flex1">
+                        <label for="sd_stability_output_format">Output Format</label>
```
Comment (Cohee1207) [inline public/scripts/extensions/stable-diffusion/settings.html]: Let's just settle on one format. I think PNG would work for everyone.

File: public/scripts/extensions/stable-diffusion/settings.html [hunk 3]
```diff
@@ -189,7 +190,82 @@
...
+                            <option value="21:9">21:9</option>
+                            <option value="2:3">2:3</option>
+                            <option value="3:2">3:2</option>
+                            <option value="4:5">4:5</option>
+                            <option value="5:4">5:4</option>
+                            <option value="9:16">9:16</option>
+                            <option value="9:21">9:21</option>
+                        </select>
+                    </div>
+                    <div class="flex1">
```
Comment (Cohee1207) [inline public/scripts/extensions/stable-diffusion/settings.html]: Could reuse an existing control for seed input.

File: public/scripts/extensions/stable-diffusion/settings.html (hunk 1 above)
Comment (WBlair1) [reply public/scripts/extensions/stable-diffusion/settings.html]: Yes, new stable image models do not use height or width—they use aspect ratio
https://platform.stability.ai/docs/api-reference#tag/Generate/paths/~1v2beta~1stable-image~1generate~1ultra/post

Screenshot_20240704-075155

Comment (Cohee1207) [reply public/scripts/extensions/stable-diffusion/settings.html]: We could normalize the aspect ratio from selected width and height by selecting the closest aspect ratio to the list as we do for OpenAI image generation (it only allows 4  specific sizes). If you don't feel like it, I can take this task on me.

Comment (WBlair1) [reply public/scripts/extensions/stable-diffusion/settings.html]: please do!!!

**Review by Cohee1207 — APPROVED** [review]

**Pull request merged**